### PR TITLE
Unity ads adapter

### PR DESCRIPTION
**Pull request by LTPhantom**
## Description

Adds a new Flutter plugin that enables the Unity Ads Mediation Adapter to use with the google_mobile_ads flutter plugin.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by luo86 — APPROVED** [review]

File: packages/mediation/gma_mediation_unity/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_unity/GmaMediationUnityPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  override fun setCCPAConsent(ccpaConsent: Boolean) {
+    val ccpaMetaData = UnityApiWrapper.createMetaData(context)
+    ccpaMetaData.set("privacy.consent", ccpaConsent)
+    ccpaMetaData.commit()
+  }
+}
+
+object UnityApiWrapper {
```
Comment (jjliu15) [inline packages/mediation/gma_mediation_unity/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_unity/GmaMediationUnityPlugin.kt:57]: Is this just needed for testing? If so, could you leave a comment explaining

Comment (jjliu15) [reply packages/mediation/gma_mediation_unity/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_unity/GmaMediationUnityPlugin.kt:57]: Also IIRC the version of mockito in Google3 does not support static mocking. Alternatively you could create a factory class that just instantiates MetaData

File: packages/mediation/gma_mediation_unity/lib/gma_mediation_unity.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:gma_mediation_unity/unity_privacy_api.g.dart';
+
+/// This class has entrypoint to call DT Exchange's SDK APIs.
```
Comment (jjliu15) [inline packages/mediation/gma_mediation_unity/lib/gma_mediation_unity.dart:17]: Unity

**Review by jjliu15 — CHANGES_REQUESTED** [review]

File: packages/mediation/gma_mediation_unity/lib/gma_mediation_unity.dart (hunk 2 above)
Comment (LTPhantom) [reply packages/mediation/gma_mediation_unity/lib/gma_mediation_unity.dart]: Ah, good catch!

File: packages/mediation/gma_mediation_unity/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_unity/GmaMediationUnityPlugin.kt (hunk 1 above)
Comment (LTPhantom) [reply packages/mediation/gma_mediation_unity/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_unity/GmaMediationUnityPlugin.kt:58]: Added the comment.

The 'advantage' we have is that our tools can (and sometimes need to) use external tools, so we can make use of the static mocking which honestly saves a lot of time of unit testing writing.

**Review by jjliu15 — APPROVED** [review]

**Pull request merged**